### ISSUE TITLE
`wrap` functions with multiple arguments

### ISSUE THREAD

**Issue by mzgubic** [invenia/TimeDag.jl#17]
```julia
julia> A = rand(iterdates())
TimeDag.Rand{Float64}(MersenneTwister(0x8cd2a3d85b05db9753173f0c2181b447))

julia> B = 2*A
TimeDag.SimpleBinary{*, Float64, TimeDag.UnionAlignment}()

julia> D = wrap((a, b, c) -> a+b-c)(A, B, A)
ERROR: MethodError: no method matching apply(::var"#111#112", ::TimeDag.Node{Float64}, ::TimeDag.Node{Float64}, ::TimeDag.Node{Float64})
Closest candidates are:
  apply(::Function, ::TimeDag.Node, ::TimeDag.Node; kwargs...) at /Users/mzgubic/JuliaEnvs/TimeDagForecasters/dev/TimeDag/src/ops/core.jl:55
  apply(::Function, ::TimeDag.Node, ::Any; kwargs...) at /Users/mzgubic/JuliaEnvs/TimeDagForecasters/dev/TimeDag/src/ops/core.jl:58
  apply(::Function, ::Any, ::TimeDag.Node; kwargs...) at /Users/mzgubic/JuliaEnvs/TimeDagForecasters/dev/TimeDag/src/ops/core.jl:59
  ...
Stacktrace:
 [1] (::TimeDag.Wrapped{var"#111#112"()})(::TimeDag.Node{Float64}, ::Vararg{TimeDag.Node{Float64}, N} where N; kwargs::Base.Iterators.Pairs{Union{}, Union{}, Tuple{}, NamedTuple{(), Tuple{}}})
   @ TimeDag ~/JuliaEnvs/TimeDagForecasters/dev/TimeDag/src/ops/core.jl:80
 [2] (::TimeDag.Wrapped{var"#111#112"()})(::TimeDag.Node{Float64}, ::Vararg{TimeDag.Node{Float64}, N} where N)
   @ TimeDag ~/JuliaEnvs/TimeDagForecasters/dev/TimeDag/src/ops/core.jl:80
 [3] top-level scope
   @ REPL[125]:1
```

Are there any inherent limitations why we cant do this? It's possible to work around it by building a tuple of arguments and then splat the tuple, but that seems awful 😁 
```julia
julia> T = wrap((a, b) -> (a, b))(A, B)
TimeDag.SimpleBinary{var"#113#114"(), Tuple{Float64, Float64}, TimeDag.UnionAlignment}()

julia> T2 = wrap((t, b) -> (t..., b))(T, A)
TimeDag.SimpleBinary{var"#115#116"(), Tuple{Float64, Float64, Float64}, TimeDag.UnionAlignment}()

julia> evaluate(wrap(t->t[1] + t[2] - t[3])(T2), DateTime(2021, 1, 1), DateTime(2021, 1, 10))
Block{Float64}(9 knots)
|                time |    value |
|            DateTime |  Float64 |
|---------------------|----------|
| 2021-01-01T00:00:00 |  1.25666 |
| 2021-01-02T00:00:00 |  1.64837 |
| 2021-01-03T00:00:00 | 0.742571 |
| 2021-01-04T00:00:00 | 0.716422 |
| 2021-01-05T00:00:00 | 0.566468 |
| 2021-01-06T00:00:00 | 0.347844 |
| 2021-01-07T00:00:00 |  1.56021 |
| 2021-01-08T00:00:00 |  1.74319 |
| 2021-01-09T00:00:00 |  0.59063 |
```

I suppose we would need to interpret the `Left` alignment as `Leftmost` alignment? I can't see any other issues - though I haven't looked very deeply.

**Comment by tpgillam** [comment]
Yep, we definitely want to support nary arguments!

The primary issue is that alignment over more than two nodes currently isn't implemented — will be easy to do this once that is done :)

**Comment by tpgillam** [comment]
I think this should be closed by #19, but feel free to re-open if you run into any other problems!

```julia
julia> x = rand(iterdates())
TimeDag.Rand{Float64}(MersenneTwister(0x6af540c0b17368c5d697e7081577efda))

julia> y = 2 * x
TimeDag.SimpleBinary{*, Float64, TimeDag.UnionAlignment}()

julia> z = wrap((a, b, c) -> a + b - c)(x, y, x)
TimeDag.SimpleNary{var"#5#6"(), 3, Float64, TimeDag.UnionAlignment}()

julia> evaluate(z, DateTime(2021), DateTime(2021, 1, 3))
Block{Float64}(2 knots)
|                time |   value |
|            DateTime | Float64 |
|---------------------|---------|
| 2021-01-01T00:00:00 | 1.89078 |
| 2021-01-02T00:00:00 | 1.92375 |
```
